internal/provisioner/plain/controllers/bundle_controller.go

```diff
@@ -464,53 +476,186 @@ func bundleImagePod(pod *corev1.Pod, source rukpakv1alpha1.ImageSource, unpackIm
 	return pod
 }
 
-func bundleGitRepoPod(pod *corev1.Pod, source rukpakv1alpha1.GitSource, unpackImage, gitClientImage string) (*corev1.Pod, error) {
-	if len(pod.Spec.InitContainers) != 2 {
-		pod.Spec.InitContainers = make([]corev1.Container, 2)
+// addUnpackerInitContainer injects the install-unpacker init container into the given pod.
+// addUnpackerInitContainer assumes the pod has an array of init containers initialized.
+func addUnpackerInitContainer(pod *corev1.Pod, unpackImage string) *corev1.Pod {
+	pod.Spec.InitContainers[0].Name = "install-unpacker"
+	pod.Spec.InitContainers[0].Image = unpackImage
+	pod.Spec.InitContainers[0].ImagePullPolicy = corev1.PullIfNotPresent
+	pod.Spec.InitContainers[0].Command = []string{"cp", "-Rv", "/unpack", "/bin/unpack"}
+	pod.Spec.InitContainers[0].VolumeMounts = []corev1.VolumeMount{{Name: "util", MountPath: "/bin"}}
+	return pod
+}
+
+func (r *BundleReconciler) getCredential(ctx context.Context, bundle *rukpakv1alpha1.Bundle) (string, string, error) {
+	if bundle.Spec.Source.Git.SecretName == "" {
+		return "", "", nil
 	}
+	secret := &corev1.Secret{}
+	err := r.Get(ctx, client.ObjectKey{Namespace: r.PodNamespace, Name: bundle.Spec.Source.Git.SecretName}, secret)
+	if err != nil {
+		return "", "", err
+	}
+	return string(secret.Data["username"]), string(secret.Data["accesstoken"]), nil
+}
 
-	pod = addUnpackerInitContainer(pod, unpackImage)
+func (r *BundleReconciler) handleGitSource(ctx context.Context, u *updater.Updater, bundle *rukpakv1alpha1.Bundle) error {
+	if bundle.Status.Phase == rukpakv1alpha1.PhaseUnpacked {
+		return nil
+	}
```

Comment:
This is functionally different than the previous code, right?

With the pod-based unpacker, we don't check the phase of the bundle before deciding to proceed with ensuring the unpack pod exists. In that scenario, even if the bundle says it is unpacked, we re-create the unpack pod, re-unpack it, and update the stored bundle (which potentially includes changes)

To get 1-to-1 behavior, is it possible to remove this early exit for now?  If I'm understanding correctly, what will happen if we remove this is:
1. First pass through, we clone the repo, build the bundle tar.gz, store it, and then update `bundle.status` with the details.
2. We run reconciliation again for this bundle because of the status update in 1. This time, we again clone the repo, build the bundle tar.gz, and store it, but the status doesn't change, so this ends the reconciliation for the original change.

We end up with 2 clones, which isn't ideal. But I think we should tackle the question of how to handle that separately. For example, I could ask "what if the status says "Unpacked", but I can see that it isn't present in my storage? I think there are a few questions along those lines that make this somewhat complex.

Reply:
I took out it.  It is working as you describe here now.